pxtcompiler/emitter/decompiler.ts

```diff
@@ -879,6 +897,33 @@ ${output}</xml>`;
                 getFieldBlock("variables_get_reporter", "VAR", varName, true), "variables_get_reporter");
         }
 
+        function mkDraggableReporterValue(valueName: string, varName: string, varType: string) {
+            const reporterType = reporterTypeForArgType(varType);
+            const reporterShadowBlock = getDraggableReporterBlock(varName, varType, true);
+            return mkValue(valueName, reporterShadowBlock, reporterType);
+        }
+
+        function reporterTypeForArgType(varType: string) {
```

Comment:
Maybe this should get refactored somewhere shared between blocklyloader and the decompiler? Seems like a lot of duplicated logic.